### ISSUE TITLE
Which is the best approach to implement a "debounce" strategy?

### ISSUE THREAD

**Issue by jorchg** [CacheControl/json-rules-engine#207]
Hello!

We are working with your library and I have to say that it is a superb engine :) Simple and powerful.
We are trying to implement somehow a "debounce" strategy, in a per-user basis:
- A user is fed to the engine and evaluate the rules.
- A rule succeed and some action is performed.
- The rule is again run against the same set of rules and the same user, but time < x has passed since last succeed and we would like that rule to not succeed.

The question is, would you implement it as part of the rule-facts-engine environment or would you implement it outside its realm? I mean, let the rule succeed every time it has to but check later if the action should be triggered or not.

We would like to implement it as part of the engine environment, which would give us a lot of flexibility in creating rules, but I don't really know if it is the correct path.

Thank you very much for your help :)
👍

**Comment by CacheControl** [comment]
At a glance, I think this could be implemented within the engine itself, with a timestamp condition that evaluates a `time-since-success` fact. For example:

```js
{
      fact: 'time-since-success',
      operator: 'greaterThan',
      value: 3600,
}
```

The `time-since-success` fact could pull the timestamp of the last success for the current user from wherever it's persisted, and calculate the time since it occurred. If a last success hasn't occurred, it would return `Infinity` (alternatively you could do an OR condition that checks for `time-since-success` to be undefined).
